module3/lessons/Week2/IntroToMVC.md

```diff
@@ -0,0 +1,307 @@
+* Provide an elegant way to create HTML output with C#.
+
+We will use Visual Studio to create a new view. Let's all follow these steps, but instead of calling our folder `HelloWorld` we will call it `Movies`.
+
+![Adding a View](/assets/images/module3/Week2/MVC_5_Create_View.png)
+
+Replace the contents of the `Views/Movies/Index.cshtml` Razor view file with the following:
+
+```C#
+<h1>Movies</h1>
+
+<p>All of the movies data will be shown here!</p>
+```
+<!-- TODO, add some content here about hot reloading. I think you we want to use the fire icon to enable reload on save? -->
+
+Navigate again to `https://localhost:<port#>/Movies`. The error should be gone, and you should now see "Movies All of the movies data will be shown here!"
+
+The `Index` method in the `MoviesController` ran the statement `return View();`, which specified that the method should use a view template file to render a response to the browser.
+
+A view template file name wasn't specified, so MVC defaulted to using the default view file. The default view has the same name as the action method, `Index` in this example. The view template `/Views/Movies/Index.cshtml` is used.
+
+## 4. Controller and View Practice
+<!-- TODO: Add practice here where students make another named route in their controller and another view with that same name. -->
+
+> With your partner: Create create a new route in your controller so that when the user visits `https://localhost:<port#>/cart` they see the string "Here are all of the items in your shopping cart"
+
+## 5. Add a Model
+
+These model classes are used with Entity Framework Core (EF Core) to work with a database.
+
+Let's start by adding a new model class.
+![Adding a model](/assets/images/module3/Week2/MVC_7_Adding_Model.png)
+
+Update the Models/Movie.cs file with the following code:
+
+```C#
+namespace MvcMovie.Models
+{
+    public class Movie
+    {
+        public int Id { get; set; }
+        public string Title { get; set; }
+        public string Genre { get; set; }
+    }
+}
+```
+The Movie class contains an Id field, which is required by the database for the primary key.
+
+### Setting up our DB
+
+Our next step is to add our `Movie` model to our database context.
+
+Update `DataAccess/MvcMovieContext` to the following:
+```C#
+using Microsoft.EntityFrameworkCore;
+using MvcMovie.Models;
+
+namespace MvcMovie.DataAccess
+{
+    public class MvcMovieContext : DbContext
+    {
+        public DbSet<Movie> Movies { get; set; }
+
+        public MvcMovieContext(DbContextOptions<MvcMovieContext> options) : base(options)
+        {
+
+        }
+    }
+}
+```
+
+<!-- TODO: is this part even neede? I think not.  
+
+public MvcMovieContext(DbContextOptions<MvcMovieContext> options) : base(options)
+        {
+
+        } 
+        -->
+
+Then we need to create our migration and apply that migration.
+Open up `Tools > NuGet Package Manager > Package Manager Console`
+```
+Add-Migration InitialCreate
+Update-Database

```

Comment:
I had to change `User` to `Username` in `appsettings.json`. I literally wasted a half-day figuring this out ☹️

Reply:
OMG I got stuck on the same thing and was also very stumped, sorry I didn't commit the change! I reached out to Megan and her second set of eyes caught it.